packages/template-retail-react-app/app/components/product-tile/index.jsx

```diff
@@ -66,18 +67,56 @@ const ProductTile = (props) => {
         ...rest
     } = props
 
-    const {currency, image, price, productId, hitType} = product
-
+    const {image, productId, hitType} = product
     // ProductTile is used by two components, RecommendedProducts and ProductList.
     // RecommendedProducts provides a localized product name as `name` and non-localized product
     // name as `productName`. ProductList provides a localized name as `productName` and does not
     // use the `name` property.
     const localizedProductName = product.name ?? product.productName
 
-    const {currency: activeCurrency} = useCurrency()
     const isFavouriteLoading = useRef(false)
     const styles = useMultiStyleConfig('ProductTile')
 
+    // NOTE: swatches will implement later to set variant accordingly,
+    // On first load, get the variant that is the represent product
+    // this is for variant product, standard/set/bundles does not have variants
+    // Also, product tile can be used in RecommendedProducts where it calls getProducts which does not have representedProduct
+    // in that case we use the first variant that has price book to set up discount price
+    // Not all variants has set in a priceBook, meaning not having tieredPrices.
+    const variant = useMemo(() => {
+        return product?.variants?.find(
+            (i) => i?.productId === product?.representedProduct?.id || !!i?.tieredPrices
+        )
+    }, [product])
+    // prioritize variant promotionalPrice over standard price
+    let currentPrice = variant
+        ? getDisplayPrice(variant)?.discountPrice || variant?.price
+        : product?.price
+
+    // check for both data returned from getProducts and productSearch
+    const isProductASet = hitType === 'set' || !!product?.type?.set
+    const isProductABundle = hitType === 'bundle' || !!product?.type?.bundle
+    const isProductAStandard = hitType === 'product' || !!product?.type?.item
```

Comment:
I'm not sure what we are doing here. Is the `hitType` not reliable on its own to determine the product type? It looks like we are saying the product is a "standard" type if it's a hit type of "product" or it has an item object? is hitType 'product' analogous to item/standard product type?

Also NIT: isProdcutASet, isProductABundle, isProductAStandard.. all but the last are kinda proper english. Maybe just stick with something like `isSet|isBundle|isStandard`